src/register-subscriptions.ts

```diff
@@ -0,0 +1,149 @@
+
+  if (!response.mnemonic) throw new Error('🛑 No mnemonic provided');
+  const registrar = ethers.Wallet.fromMnemonic(response.mnemonic);
+
+  // Get all the chains the API will be deployed on
+  const apiChains = [
+    ...new Set(Object.values(apiData.beacons).flatMap((beacon) => beacon.chains.map((chain) => chain.name))),
+  ];
+
+  // Build NounceManagers for each chain
+  const nonceManagers = apiChains.reduce(
+    (acc, chainName) => ({
+      ...acc,
+      [chainName]: new NonceManager(
+        registrar.connect(new ethers.providers.JsonRpcProvider(chainNameToPublicRPCUrl[chainName]))
+      ),
+    }),
+    {} as { [chainName: string]: NonceManager }
+  );
+
+  const subscriptionPromises = Object.entries(apiData.beacons).flatMap(([beaconName, beacon]) => {
+    const DapiServerInteface = DapiServerInterface();
+    const parameters = '0x';
+    const airnodeAddress = beacon.airnodeAddress;
+    const templateId = beacon.templateId;
+    const threshold = ethers.BigNumber.from(100000000)
+      .mul(beacon.updateConditionPercentage * 100)
+      .div(10000);
+    const beaconUpdateSubscriptionConditionParameters = ethers.utils.defaultAbiCoder.encode(['uint256'], [threshold]);
+    const encodedBeaconUpdateSubscriptionConditions = encode([
+      {
+        type: 'bytes32',
+        name: '_conditionFunctionId',
+        value: ethers.utils.defaultAbiCoder.encode(
+          ['bytes4'],
+          [DapiServerInteface.getSighash('conditionPspBeaconUpdate')]
+        ),
+      },
+      { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
+    ]);
+    return beacon.chains.map(async (chain) => {
+      const chainId = chainNameToChainId[chain.name];
+      if (!chainId) throw new Error(`🛑 Unknown chain name: ${chain.name}`);
+
+      if (!chainNameToPublicRPCUrl[chain.name]) throw new Error(`🛑 No public RPC URL for chain ${chain.name}`);
+      const provider = new ethers.providers.JsonRpcProvider(chainNameToPublicRPCUrl[chain.name]);
+      const DapiServerAddress = operationsRepository.documentation.chains[chain.name].contracts['DapiServer'];
+      if (!DapiServerAddress) throw new Error(`🛑 No DapiServer contract address for chain ${chain.name}`);
+      const DapiServer = DapiServerContract(DapiServerAddress, provider);
+
+      const sponsor = chain.sponsor;
+
+      const expectedSubscriptionId = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(
+          ['uint256', 'address', 'bytes32', 'bytes', 'bytes', 'address', 'address', 'address', 'bytes4'],
+          [
+            chainId,
+            airnodeAddress,
+            templateId,
+            parameters,
+            encodedBeaconUpdateSubscriptionConditions,
+            airnodeAddress,
+            sponsor,
+            DapiServerAddress,
+            DapiServerInteface.getSighash('fulfillPspBeaconUpdate'),
+          ]
+        )
+      );
+
+      // eslint-disable-next-line no-async-promise-executor
```

Comment:
I don't think its really a problem, in particular because we catch the error in the try-catch block

Reply:
In general the PR is neat and your work has improved a lot :)
...but you shouldn't need that eslint ignore flag :stuck_out_tongue: 
There are scenarios where an ignore flag is necessary, but it's usually somewhere _weird_, like if you're dealing with low level functionality in Node.

Node has `sync` and `async` flows and `Promise` is like a converter/interface layer between them. The reason you had to use this ignore tag was because you're trying to pass an `async` function to a `Promise`, but an `async` function by it's nature is already a Promise; an `async` function returns a Promise<Type>, like Promise<boolean>.

There are two/three different ways you could have handled this section:
These flows underneath do almost the same thing because a lot of the differences are "syntactic sugar".
1.
You can take away the Promise entirely:
```
      try {
        console.log(`🔗 Registering subscriptionId for beacon ${beaconName} on chain ${chain.name}`);

        const registerBeaconUpdateSubscription = DapiServer.connect(
          nonceManagers[chain.name],
        ).registerBeaconUpdateSubscription(
          airnodeAddress,
          templateId,
          encodedBeaconUpdateSubscriptionConditions,
          airnodeAddress,
          sponsor,
        );

        // Check that the transaction is complete
        const tx = registerBeaconUpdateSubscription.wait();

        const subscriptionId = tx.events.find(
          (event: { event: string }) => event.event === 'RegisteredBeaconUpdateSubscription',
        ).args.subscriptionId;

        if (subscriptionId !== expectedSubscriptionId) {
          console.error(`🛑 The subscription ID ${subscriptionId} does not match the expected ID ${expectedSubscriptionId}`);
          return;
        }

        console.log(
          `✅ Subscription registered with ID ${subscriptionId} for beacon ${beaconName} on chain ${chain.name}`);

      } catch (error) {
        console.error(`🛑 Error registering subscription for beacon ${beaconName} on chain ${chain.name}`);
      }
```
2. You could build something synchronous:
```
      return new Promise((resolve, reject) => {
        try {
          console.log(`🔗 Registering subscriptionId for beacon ${beaconName} on chain ${chain.name}`);

          const registerBeaconUpdateSubscription = DapiServer.connect(
            nonceManagers[chain.name]
          ).registerBeaconUpdateSubscription(
            airnodeAddress,
            templateId,
            encodedBeaconUpdateSubscriptionConditions,
            airnodeAddress,
            sponsor
          )
            .catch(reject)
            .then(() => {
            // Check that the transaction is complete
            const tx = registerBeaconUpdateSubscription.wait().then(() => {
              const subscriptionId = tx.events.find(
                (event: { event: string }) => event.event === 'RegisteredBeaconUpdateSubscription'
              ).args.subscriptionId;

              if (subscriptionId !== expectedSubscriptionId) {
                reject(`🛑 The subscription ID ${subscriptionId} does not match the expected ID ${expectedSubscriptionId}`);
              }

              resolve(
                `✅ Subscription registered with ID ${subscriptionId} for beacon ${beaconName} on chain ${chain.name}`
              );
            });
          });
        } catch (error) {
          console.error(`🛑 Error registering subscription for beacon ${beaconName} on chain ${chain.name}`);
          reject(error);
        }
      });
```
3. Ain't no-one got time for that.

I hope this helps. It took me a while to get it too.

Reply:
Thanks for the through explanation, I also dug a bit deep into what is happening and I realized that what I was doing isn't needed. When declaring `Array.map(async () => {})` it is implicit that the return type will be an array of promises. This means any code we execute in this mapped function will be an asyncronous promise that can be resolved later using `promise.all`. Explicitly declaring a return type of promise isn't needed in this case.

I did a variation of 1. 2 is esentially syntatic sugar, its not acutally synchrounous because `.then` is [just another variation of await ](https://stackoverflow.com/a/54497100) except its alot less readable.

Reply:
I prefer 1 :) All good